cmd/kubetype-gen/generators/types.go

```diff
@@ -61,6 +61,12 @@ func (g *typesGenerator) GenerateType(c *generator.Context, t *types.Type, w io.
 		"IstioStatus": c.Universe.Type(types.Name{Name: "IstioStatus", Package: "istio.io/api/meta/v1alpha1"}),
 	}
 	for _, kubeType := range kubeTypes {
+		localM := m
+		// ServiceEntry has a unique status type which includes addresses for auto allocated IPs, substitute IstioServiceEntryStatus
+		// for IstioStatus when type is ServiceEntry
+		if kubeType.Type().Name.Name == "ServiceEntry" {
```

Comment:
would be ideal if this was from checking some comment on the type indicating this

Reply:
potential impl in latest commit, thanks for the suggestion, I think this would allow decoupling this PR from API PR somewhat, yeah?